lib/thrift/generator/struct_binary_protocol.ex

```diff
@@ -769,30 +769,55 @@ defmodule Thrift.Generator.StructBinaryProtocol do
     end
   end
 
-  defp field_serializer(%Field{name: name, type: :bool, id: id}, _file_group) do
-    var = Macro.var(name, nil)
+  defp field_serializer(%Field{name: name, type: :bool, id: id, required: false}, _file_group) do
     quote do
-      case unquote(var) do
+      case unquote(Macro.var(name, nil)) do
         nil ->
           <<>>
         false ->
           <<unquote(@bool), unquote(id) :: size(16), 0>>
+        true ->
+          <<unquote(@bool), unquote(id) :: size(16), 1>>
         _ ->
+          raise Thrift.MissingFieldException,
```

Comment:
Warning: Pedantry.
We're raising a MissingFieldException, but the field isn't technically missing; it's there, but with an invalid field. Consider an `InvalidFieldValueException` instead?